compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/intrinsic.rs

```diff
@@ -829,4 +836,47 @@ impl<'tcx> GotocCtx<'tcx> {
             loc,
         )
     }
+
+    /// simd_shuffle constructs a new vector from the elements of two input vectors,
+    /// choosing values according to an input array of indexes.
+    ///
+    /// This code mimics CBMC's `shuffle_vector_exprt::lower()` here:
+    /// https://github.com/diffblue/cbmc/blob/develop/src/ansi-c/c_expr.cpp
+    ///
+    /// We can't use shuffle_vector_exprt because it's not understood by the CBMC backend,
+    /// it's immediately lowered by the C frontend.
+    /// Issue: https://github.com/diffblue/cbmc/issues/6297
+    pub fn codegen_intrinsic_simd_shuffle(
+        &mut self,
+        mut fargs: Vec<Expr>,
+        p: &Place<'tcx>,
+        cbmc_ret_ty: Type,
+        n: u64,
+    ) -> Stmt {
+        assert!(fargs.len() == 3, "simd_shuffle had unexpected arguments {:?}", fargs);
+        // vector, size n: translated as vector types which cbmc treats as arrays
+        let vec1 = fargs.remove(0);
+        let vec2 = fargs.remove(0);
+        // [u32; n]: translated wrapped in a struct
+        let indexes = fargs.remove(0);
+
+        // An unsigned type here causes an invariant violation in CBMC.
+        // Issue: https://github.com/diffblue/cbmc/issues/6298
+        let st_rep = Type::ssize_t();
```

Comment:
It might be clearer to just use `Type::ssize_t()` wherever this is used.

Reply:
I had that thought too, but I felt like keeping it since it created a place for the comment about the CBMC issue.